### PR TITLE
Feature/nijo 960 handling not found status

### PR DESCRIPTION
**Pull request by mingazhev**
1. Changed EDO booking to use it's own enums for booking states. In plans to merge Pending and WaitingForResponse states, 
change InternalProcessing to Created and some other improvements.
2. Added a special state "Needs manual correction" and setting it if booking has "NotFound" state more than 30 minutes.
Later this should be used to notify administrators on that bookings.

### CODE REVIEW

**Review by kirillta — APPROVED** [review]
Approved with wording comments in mind

File: Api/Services/Accommodations/Bookings/BookingRecordsManager.cs [hunk 1]
```diff
@@ -102,7 +99,7 @@ async Task<Data.Booking.Booking> Create((string itn, string referenceCode) tags)
         public async Task UpdateBookingDetails(EdoContracts.Accommodations.Booking bookingDetails, Data.Booking.Booking booking)
         {
             booking.SupplierReferenceCode = bookingDetails.AgentReference;
-            booking.Status = bookingDetails.Status;
+            booking.Status = bookingDetails.Status.ToStatus();
```
[the remark this answers is not in the document]
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/BookingRecordsManager.cs]: Fixed

File: Api/Services/Accommodations/Bookings/BookingRecordsManager.cs [hunk 2]
```diff
@@ -264,6 +261,15 @@ async Task SetPaymentMethod(Booking booking)
         }
 
 
+        public async Task SetNeedsManualCorrection(Data.Booking.Booking booking)
```
[the remark this answers is not in the document]
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/BookingRecordsManager.cs:264]: Fixed

File: Api/Services/Accommodations/Bookings/BookingService.cs [hunk 3]
```diff
@@ -292,14 +292,21 @@ void WriteLogFailure(ProblemDetails problemDetails)
...
 
             await _bookingAuditLogService.Add(bookingResponse, booking);
 
             _logger.LogBookingResponseProcessStarted(
                 $"Start the booking response processing with the reference code '{bookingResponse.ReferenceCode}'. Old status: {booking.Status}");
 
+            if (bookingResponse.Status == BookingStatusCodes.NotFound)
+            {
+                await ProcessBookingNotFound();
+                return;
```
[the remark this answers is not in the document]
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/BookingService.cs:306]: No it couldn't. Does not compile

File: HappyTravel.Edo.Common/Enums/BookingStatuses.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        // TODO: Remove not needed statuses and migrate existing bookings statuses
+        InternalProcessing = 0,
+        WaitingForResponse = 1,
+        Pending = 2,
+        Confirmed = 3,
+        Cancelled = 4,
+        Rejected = 5,
+        Invalid = 6,
+        Reverted = 7,
+        NeedsManualCorrection = 8
```
[the remark this answers is not in the document]
Comment (mingazhev) [reply HappyTravel.Edo.Common/Enums/BookingStatuses.cs:18]: Fixed

**Review by hasmikharutyunyan — APPROVED** [review]

**Review by oleg1std — APPROVED** [review]

**Pull request merged**